src/Akka.HealthCheck.Persistence/AkkaPersistenceLivenessProbe.cs

```diff
@@ -154,8 +159,12 @@ private bool HandleSubscriptions(object msg)
         private void HandleRecoveryStatus(PersistenceLivenessStatus livenessStatus)
         {
             if(_logInfo)
-                _log.Debug("Received recovery status {0} from probe.", livenessStatus);
+                _log.Debug("Received recovery status {0} from probe. First attempt? {1}", livenessStatus, livenessStatus.Warmup);
+            
             _currentLivenessStatus = livenessStatus;
+            if (livenessStatus.Warmup && (!livenessStatus.SnapshotSaved || !livenessStatus.JournalPersisted))
+                _firstIndex++;
```

Comment:
So this value goes up each time the warmup probe recovers without knowing whether or not the two persist operations have both completed successfully?

Reply:
Warmup is a property we pass back to the probe that the suicide actor is in "first attempt" mode, if any of the snapshot save or journal persist failed, we extend the "first attempt" time to the next cycle until both succeed. 

This is because the liveness status does not allow for undefined/null recovery status for the probe to detect if recovery was performed on both snapshot and journal.

Maybe we should change the recovery properties of the persistence liveness status to be nullable instead?

Reply:
> Maybe we should change the recovery properties of the persistence liveness status to be nullable instead?

Yes, if there's a better / more intuitive way of modeling this we should do that instead.